src/ast/passes/semantic_analyser.cpp

```diff
@@ -1790,7 +1790,8 @@ void SemanticAnalyser::visit(Binop &binop)
     return;
   }
 
-  if (lht.IsIntTy() && rht.IsIntTy()) {
+  if ((lht.IsCountTy() || lht.IsSumTy() || lht.IsIntTy()) &&
+      (rht.IsCountTy() || rht.IsSumTy() || rht.IsIntTy())) {
```

Comment:
I'm wondering if we can be more generic here. For example, this doesn't quite fix Jesper's issue where he wanted to printf() the count() value. 

Perhaps we can allow casts to int64/int32/etc from Type::count? I kinda like the idea of explicit casts cuz it'll at least hint we're gonna do something magical. Eg.

```
@ = count();
printf("%d\n", (int64)@);
```

WDYT?

Reply:
Yeah, I like explicit casts - let me look into supporting that.

Reply:
Added this in the latest version.